### PR TITLE
Changed hash computation for Pydantic2 and ForwardRef

### PR DESCRIPTION
**Pull request by teo-milea**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/pr-tests-hagrid.yml [hunk 1]
```diff
@@ -80,7 +80,7 @@ jobs:
         if: steps.changes.outputs.hagrid == 'true'
         run: |
           bandit -r hagrid
-          safety check -i 42923 -i 54229 -i 54230 -i 54230 -i 54229 -i 62044 -i 65213
+          safety check -i 42923 -i 54229 -i 54230 -i 54230 -i 54229 -i 62044 -i 65213 -i 54564
```
Comment (shubham3121) [inline .github/workflows/pr-tests-hagrid.yml:83]: 💯 👍🏾

File: notebooks/api/0.8/10-container-images.ipynb [hunk 2]
```diff
@@ -131,8 +131,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "custom_dockerfile_str = f\"\"\"\n",
-    "FROM openmined/grid-backend:{syft_base_worker_tag}\n",
+    "custom_dockerfile_str = \"\"\"\n",
+    "FROM openmined/grid-backend:0.8.5-beta.10\n",
```
Comment (khoaguin) [inline notebooks/api/0.8/10-container-images.ipynb:135]: we are hard coding this here just for this release specifically, right?

File: packages/syft/src/syft/node/node.py [hunk 3]
```diff
@@ -1427,8 +1424,6 @@ def get_unauthed_context(
         return UnauthedServiceContext(node=self, login_credentials=login_credentials)
 
     def create_initial_settings(self, admin_email: str) -> NodeSettingsV2 | None:
-        if self.name is None:
```
Comment (rasswanth-s) [inline packages/syft/src/syft/node/node.py:1430]: Q: Do random name creation cause bugs?

Comment (shubham3121) [reply packages/syft/src/syft/node/node.py:1430]: Nope, we're creating random_name in __init__ , this statement was unreachable.

**Review by shubham3121 — APPROVED** [review]
💯 🔥

**Pull request merged**